httpgrpc/server/server.go

```diff
@@ -27,6 +27,10 @@ import (
 	"github.com/grafana/dskit/middleware"
 )
 
+var (
+	DoNotLogErrorHeaderKey = http.CanonicalHeaderKey("X-DoNotLogError")
```

Comment:
Can we add short doc saying how to use this header? If it's present in the response (with any value), and response has status code 5xx, then generated error will be marked to not be logged.

Reply:
I have added a doc, and I have updated CHANGELOG.md.